### PR TITLE
Plugins guide (fixes #40)

### LINKED ISSUE #40: Plugins guide

### PR DESCRIPTION
**Pull request by ehborisov**

### CODE REVIEW

File: docs/2.0/plugins.adoc [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+To be able to use the API you should simply download the `allure-plugin-api` dependency
+link:https://mvnrepository.com/artifact/io.qameta.allure/allure-plugin-api[from] the jcenter repository.
+To do so add to your project build script:
+
+in Gradle:
+
+[source, groovy]
+----
+dependencies {
+    compile('io.qameta.allure:allure-plugin-api:${allureVersion}')
```
Comment (baev) [inline docs/2.0/plugins.adoc]: compileOnly

File: docs/2.0/plugins.adoc [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+}
+----
+
+in Maven:
+[source, xml]
+----
+        <dependency>
+            <groupId>io.qameta.allure</groupId>
+            <artifactId>allure-plugin-api</artifactId>
+            <version>${allureVersion}</version>
```
Comment (baev) [inline docs/2.0/plugins.adoc]: add provided scope

File: docs/2.0/plugins.adoc [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+}
+
+apply plugin: 'java-library-distribution'
+
+jar {
+    archiveName = 'plugin.jar'
+}
+
+dependencies {
+    compile('io.qameta.allure:allure-plugin-api:2.0-BETA8')
```
Comment (baev) [inline docs/2.0/plugins.adoc]: compileOnly

File: docs/2.0/plugins.adoc [hunk 4]
```diff
@@ -187,11 +243,265 @@ We will cover all other classes by examples down below.
...
+
+After all that add `addTab` function call would look like this:
+
+[source, javascript]
+----
+allure.api.addTab('mytab', {
+    title: 'My Tab', icon: 'fa fa-trophy',
+    route: 'mytab',
+    onEnter: (function () {
+        return new MyLayout({name: 'My Tab', text: 'Hello World!'})
```
Comment (baev) [inline docs/2.0/plugins.adoc]: did you mean `MyView`?

**Comment by eroshenkoam** [comment]
https://gohugo.io/overview/quickstart/#step-2-scaffold-bookshelf-hugo-site

**Comment by baev** [comment]
@qameta-ci retest this please

**Pull request merged**